plugins/nodes/src/js/components/NodesTableWrapper.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from "react";
+import graphqlObservable from "reactive-graphql";
+import gql from "graphql-tag";
+import { Observable, combineLatest } from "rxjs";
+import { map, distinctUntilChanged, switchMap } from "rxjs/operators";
+import { componentFromStream } from "@dcos/data-service";
+import { TableProps } from "@dcos/ui-kit/dist/packages/table/components/Table";
+
+import { NodeNetwork } from "../data/NodesNetworkClient";
+import { default as schema } from "../data/NodesResolver";
+import Node from "#SRC/js/structs/Node";
+
+interface TableWrapperProps {
+  data: any[];
+  children: React.ReactElement<TableProps>;
+}
+
+const getGraphQl = (privateIP: string) => {
+  return graphqlObservable<{ node: { network: NodeNetwork } }>(
+    gql`
+      query {
+        node(privateIP: $privateIP) {
+          network {
+            public_ips
+          }
+        }
+      }
+    `,
+    schema,
+    { privateIP }
+  ).pipe(map(response => response.data));
+};
+// Put in the same props as you would in the NodesTable
+const NodesTableWrapper = componentFromStream<TableWrapperProps>(props$ => {
+  const data$ = (props$ as Observable<TableWrapperProps>).pipe(
+    map((props: TableWrapperProps) => props.data),
+    distinctUntilChanged((x, y) => {
+      if (x.length === y.length) {
+        return JSON.stringify(x) === JSON.stringify(y);
```

Comment:
This seems pretty strange.  It appears like this will run quite often, even just hovering over the rows, as the table grows this seems like a potential slow down...  Is there way to use some kind of immutable so we could a reference equal?  Would changing the approach as you mentioned in your note avoid this?

Reply:
Well the problem here is we are using Node structs on the top level by that making the comparison already difficult.  In the most cases we are avoiding a much more complex computation by unpacking each node struct object adding the network information from the endpoints and then repacking it into node structs. Also I feel adding immutable.js to dcos-ui and changing the nodes table to work with a immutable list feels to me a bit of a scope creep. This is not nice but unfortunately the whole nodes page is not nice. We should plan to do a bigger refactoring of the whole nodes page. However this pattern is not here to stay.

Reply:
`complex computation by unpacking each node struct object adding the network information from the endpoints and then repacking it into node structs` is that a one time cost?  Can you run a quick test where you load that table up with say 200 nodes and see how it performs, scrolling + hovering and what not?

Reply:
I checked against a bigger cluster the performance was not a lot worse then without the change. I however think we should remove it just to make sure also by sharing the connection we are only sending one request. how do you feel about this @juliangieseke who suggested this check in the first place?